lib/certificate.js

```diff
@@ -0,0 +1,162 @@
+import sqlite3 from 'sqlite3';
+import Promise from 'bluebird';
+import openssl from 'openssl-wrapper';
+import crypto from 'crypto';
+
+let tset = `<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n
+    <!DOCTYPE plist PUBLIC \"-//Apple//DTD PLIST 1.0//EN\" \"http://www.apple.com/DTDs/PropertyList-1.0.dtd\">
+    <plist version=\"1.0\">
+    <array/>
+</plist>`;
+
+let _promiseCb = (resolve, reject) => {
+  return (err, result) => {
+    if (err) {
+      reject(err);
+    } else {
+      resolve(result);
+    }
+  };
+};
+
+/**
+ * Library for programatically adding certificates 
+ */
+class Certificate {
+
+  constructor (pemFilename){ 
+    this.pemFilename = pemFilename;
+  }
+
+  /**
+   * Add a certificate to the TrustStore
+   */
+  async add (dir) {
+    let data = await this.getDerData(this.pemFilename);
+    let subject = await this.getSubject(this.pemFilename);
+    let fingerprint = await this.getFingerPrint(this.data);
+
+    let trustStore = new TrustStore(dir);
+    return trustStore.addRecord(fingerprint, tset, subject, data);
+  }
+
+  /**
+   * Checks if keychain at given directory has this certificate
+   */
+  async has (dir) {
+    let subject = await this.getSubject(this.pemFilename);
+
+    let trustStore = new TrustStore(dir);
+    let records = await trustStore.getRecords(subject);
+    return records.length > 0;
+  }
+
+  /**
+   * Remove certificate from the TrustStore
+   */
+  async remove (dir) { 
+    let subject = await this.getSubject(this.pemFilename);
+    let trustStore = new TrustStore(dir);
+    return trustStore.removeRecord(subject);
+  }
+
+  /**
+   * Translate PEM file to DER buffer
+   */
+  async getDerData (){
+    if (this.data)
```

Comment:
we always use braces, even for one-line `if` statements. the only exception was in ES5 code (`if (err) return cb(err);`)